### PR TITLE
Create clinic database based on a diagram

### PR DESCRIPTION
**Pull request by dfsalam**
# In this pull request we:
- ### Created clinic database;
- ### Created tables patients, invoices, medical_histories, invoice_items, and treatments.
- ### Created the join table procedures.
- ### Added the foreign keys for all tables.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+ALTER TABLE invoices
+ADD CONSTRAINT fk_medical_histories
+  FOREIGN KEY (medical_history_id)
+  REFERENCES medical_histories(id);
+
+CREATE TABLE procedures (
+  id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  medical_histories_id INTEGER,
+  treatment_id INTEGER
```
Comment (christianbiring1) [inline schema_based_on_diagram.sql]: - [x] Great on creating a join table to allow the `many-to-many` relationship :+1:

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+ALTER TABLE procedures
+ADD CONSTRAINT fk_medical_histories
+  FOREIGN KEY (medical_histories_id)
+  REFERENCES medical_histories(id);
+
+ALTER TABLE procedures
+ADD CONSTRAINT fk_treatments
+  FOREIGN KEY (treatment_id)
+  REFERENCES treatments(id);
```
Comment (christianbiring1) [inline schema_based_on_diagram.sql:53]: [OPTIONAL]
- [ ] The join-table can be shortened like this :smile: 
```sql
  CREATE TABLE procedures (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @dfsalam @luis-pomare , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 
soclose

## Highlights
- [x] Great on implementing the db schema and the many-to-many relationship :+1: 
- [x] Descriptive PR `title` and `summary` :+1: 
## Required Changes ♻️
- [ ] As required in the project requirements, Kindly make sure all the foreign_keys have indexes.
`Hint: You may consider saying:
```sql
CREATE INDEX ON medical_histories (patient_id);
```
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (dfsalam) [reply schema_based_on_diagram.sql:53]: Done. We refactor the table creation and create all the indexes.

**Review by OyePriscilla — APPROVED** [review]
Hi @dfsalam and @luis-pomare 👏 👏 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Used CREATE TABLE command to create all tables according to the diagram ✔️
Implement join table for the many-to-many relationships ✔️
Made the required changes ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**